src/Symfony/Component/JsonPath/JsonCrawlerInterface.php

```diff
@@ -22,10 +22,12 @@
 interface JsonCrawlerInterface
 {
     /**
+     * @param string|resource $data The JSON string or a stream resource containing the JSON string
+     *
      * @return list<array|string|float|int|bool|null>
      *
      * @throws InvalidArgumentException When the JSON string provided to the crawler cannot be decoded
      * @throws JsonCrawlerException When a syntax error occurs in the provided JSON path
      */
-    public function find(string|JsonPath $query): array;
+    public function find(string|JsonPath $query, mixed $data): array;
```

Comment:
How did do you decides of the order of the arguments?
I think we should put `$data` before `$query` to be consistent with string methods that put the data (haystack) before the query.

Reply:
It sticks to array functions order, because in the end, it allows to search in collections. But both order suit me fine honestly 🙂

Reply:
I tend to put loosely-typed parameters last, but Jérôme's reasoning sounds good